fuse-overlayfs.1.md

```diff
@@ -32,9 +32,58 @@ done to the file system will be written.
 A directory used internally by fuse-overlays, must be on the same file
 system as the upper dir.
 
+**-o workdir=workdir**
+A directory used internally by fuse-overlays, must be on the same file
+system as the upper dir.
+
+**-o uidmapping=UID:MAPPED-UID:LEN[,UID2:MAPPED-UID2:LEN2]**
+**-o gidmapping=GID:MAPPED-GID:LEN[,GID2:MAPPED-GID2:LEN2]**
+Specifies the dynamic UID/GID mapping used by fuse-overlayfs when
+reading/writing files to the system.
+
+It is useful to share the same storage among different user namespaces
+and counter effect the mapping done by the user namespace itself, and
+without requiring to chown the files.
+
+For example, given on the host two files like:
+
+$ stat -c %u:%g lower/a lower/b
+0:0
+1:1
+
+When we run in a user namespace with the following configuration:
+$ cat /proc/self/uid_map
```

Comment:
Is it possible for fuse-overlayfs to read this itself instead of asking the user to pass it in in options?  Are the options just for overriding a default?

Reply:
these are not really supposed to be used manually, the only user I am aware of right now is `containers/storage` so used internally by tools like Podman/Buildah/CRI-O.

Also, fuse-overlayfs itself is running outside of the namespace, so it is not enough to read `/proc/self/uid_map`

Reply:
Ok, very interesting. I am thinking about how to use it as part of a [proposed Singularity feature for rootless containers](https://github.com/sylabs/singularity/pull/2718) along with kernel 4.18 or later.  I thought fuse-overlayfs would be used inside the namespace but I guess I don't yet fully understand it.